sympy/physics/hydrogen.py

```diff
@@ -168,9 +168,13 @@ def E_nl(n, Z=1):
 
     """
     n, Z = S(n), S(Z)
-    if n.is_integer and (n < 1):
+    if n.is_constant() and n.is_integer == True and (n < 1):
         raise ValueError("'n' must be positive integer")
-    return -Z**2/(2*n**2)
+    elif n.is_integer == True:
+        return -Z**2/(2*n**2)
+    else:
+        raise ValueError("'n' must be positive integer")
```

Comment:
This looks like it would raise if the symbol is not explicitly declared as integer. That's not something that happens in most of sympy where symbols are usually assumed valid for the context they are used. Is that a common convention in this part of the codebase?

Reply:
I thought if the symbol is explicitly declared as an integer, it would be more accurate and therefore didn't make any assumptions, although I believe that what you are saying seems more logical. Should I make changes so that it accepts symbols that are explicitly declared as integers?

Reply:
With the changes here what happens if I call this with `n = Symbol('n')`?

Reply:
It raises `ValueError("'n' must be positive integer") ` But returns `-Z**2/(2*n**2)` with `n = Symbol('n', integer=True)`

Reply:
And without this PR what would happen?

Reply:
Without this PR, the code works for `n = Symbol('n') ` but shows an error with `n = Symbol('n', integer=True) ` when compared to 1.